analysis/dataset_definition_med_status_data_development.py

```diff
@@ -0,0 +1,124 @@
+from ehrql import create_dataset, months, days
+from ehrql.tables.tpp import patients, clinical_events
+from ehrql.tables.raw.tpp import medications
+
+from datetime import date
+
+from codelists import *
+
+dataset = create_dataset()
+dataset.define_population(patients.exists_for_patient())
+
+# Pharmacy first clinical codes
+pharmacy_first_event_codes = [
+    "1659111000000107",
+    "1659121000000101",
+    "1577041000000109",
+    "983341000000102",
+]
+
+# Combine all medication codelists
+pharmacy_first_med_codes = (
+    acute_otitis_media_cod
+    + first_impetigo_treatment_cod
+    + infected_insect_bites_cod
+    + shingles_treatment_cod
+    + sinusitis_cod
+    + sore_throat_cod
+    + urinary_tract_infection_cod
+)
+
+# https://www.england.nhs.uk/primary-care/pharmacy/pharmacy-first/
+pharmacy_first_launch_date = date(2024, 1, 31)
```

Comment:
This is a specific day but generally I think OpenSAFELY designed with months in mind? I'm not sure if this will include all of Jan as pre- or post- launch? Or will it split each month in an unusual way (months running 31st Jan - 27th Feb???).
Might be much simpler to take Feb 24 as start date (none of the BSA datasets have any Pharmacy First data for Jan 24 - might be a pragmatic decision, or maybe there was nothing on 31st Jan).